bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactory.java

```diff
@@ -0,0 +1,278 @@
+                            commonHttpClientFacade = new HttpClientDelegate(commonHttpClient);
+                            logger.info("jetty shared http client created");
+                        }
+
+                        return null;
+                    }
+                });
+            } catch (PrivilegedActionException e) {
+                Throwable cause = e.getCause();
+                if (cause instanceof RuntimeException) {
+                    throw (RuntimeException) cause;
+                } else {
+                    throw new RuntimeException(cause);
+                }
+            }
+        }
+    }
+
+    private HttpClient createHttpClientInternal(String consumerName, String endpoint) {
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<HttpClient>() {
+                @Override
+                public HttpClient run() {
+
+                    logger.info("creating httpClient for endpoint {}", endpoint);
+                    SslContextFactory sslContextFactory = new SslContextFactory();
+                    sslContextFactory.setEndpointIdentificationAlgorithm("HTTPS");
+                    if (endpoint != null && trustmanagerProvider != null) {
+                        Stream<TrustManager> trustManagerStream = trustmanagerProvider.getTrustManagers(endpoint);
+                        TrustManager[] trustManagers = trustManagerStream.toArray(TrustManager[]::new);
+                        if (trustManagers.length > 0) {
+                            logger.info(
+                                    "using custom trustmanagers (certificate pinning) for httpClient for endpoint {}",
+                                    endpoint);
+                            try {
+                                SSLContext sslContext = SSLContext.getInstance("TLS");
+                                sslContext.init(null, trustManagers, null);
+                                sslContextFactory.setSslContext(sslContext);
+                            } catch (Exception ex) {
+                                throw new RuntimeException(
+                                        "Cannot create an TLS context for the endpoint '" + endpoint + "'!", ex);
+                            }
+                        }
+                    }
+                    String excludeCipherSuites[] = { "^.*_(MD5)$" };
+                    sslContextFactory.setExcludeCipherSuites(excludeCipherSuites);
+
+                    HttpClient httpClient = new HttpClient(sslContextFactory);
+                    final QueuedThreadPool queuedThreadPool = createThreadPool(consumerName, minThreadsCustom,
+                            maxThreadsCustom, keepAliveTimeoutCustom);
+
+                    httpClient.setExecutor(queuedThreadPool);
+
+                    try {
+                        httpClient.start();
+                    } catch (Exception e) {
+                        logger.error("Could not start jetty client", e);
```

Comment:
Are exceptions logged on creation? Otherwise the error will not end up in the log, if someone catches this exception without logging.

Reply:
https://stackoverflow.com/questions/6639963/why-is-log-and-throw-considered-an-anti-pattern

> Otherwise the error will not end up in the log, if someone catches this exception without logging.

which is its own anti-pattern ;-)